### PR TITLE
Add custom sorting support and date formatters. Resolves #51

### LINKED ISSUE #51: Improve DateRanges postprocessor month formatting

### PR DESCRIPTION
**Pull request by adrian-gomez**
@bilby91 @andres99x Once this is merged it would be used like this on the date field that is grouped by
```
class UsersReport < Noventius::Report
  include Noventius::Extension::DateQuery
  include Noventius::Extension::DateFormat

  column :created_at, -> { date_component_type(group_by) },
                                   value: ->(row) { format_date_component_value(group_by, row[0]) },
                                   sort_value: ->(row) { date_component_sort_value(group_by, row[0]) }
end
```
or a helper method could be created if this is used in many places:
```
class BaseReport < Noventius::Report
  include Noventius::Extension::DateQuery
  include Noventius::Extension::DateFormat

  def self.grouped_date_column(name, group_method, index_or_name)
    column name, -> { date_component_type(public_send(group_method)) },
           value: ->(row) { format_date_component_value(public_send(group_method), row[index_or_name]) },
           sort_value: ->(row) { date_component_sort_value(public_send(group_method), row[index_or_name]) }
  end
end

class UsersReport < BaseReport
  grouped_date_column :created_at, :group_by, 3
end
```

### CODE REVIEW

File: lib/noventius/column.rb [hunk 1]
```diff
@@ -4,37 +4,44 @@ class Column
...
     end
 
-    def value(report, row)
-      if @value.is_a?(Proc)
-        report.instance_exec(row, &@value)
-      elsif @value.is_a?(Symbol)
-        report.public_send(@value, row)
+    def type(report)
+      if @type.is_a?(Proc)
+        report.instance_exec(&@type)
```
Comment (bilby91) [inline lib/noventius/column.rb]: `method_or_proc` here ?

**Comment by bilby91** [comment]
Please @adrian-gomez, change destination branch to master +1

File: lib/noventius/column.rb (hunk 1 above)
Comment (adrian-gomez) [inline lib/noventius/column.rb]: This one is only proc, when a symbol is given it has to be one of the supported types

**Comment by adrian-gomez** [comment]
@bilby91 I'm going to open a new one for master you can delete this one when that is merged

**Comment by bilby91** [comment]
We need a rebase now :(

**Comment by adrian-gomez** [comment]
rebased

**Pull request merged**